pkg/certificate/mrc_reconciler.go

```diff
@@ -80,6 +64,32 @@ func getSigningAndValidatingMRCs(mrcList []*v1alpha2.MeshRootCertificate) (*v1al
 	return mrcList[0], mrcList[1], nil
 }
 
+// ValidateMRCCombination takes a list of MRCs and ensures that the MRC combination is valid
+func ValidateMRCCombination(mrcList []*v1alpha2.MeshRootCertificate) error {
+	if len(mrcList) == 0 {
+		log.Error().Err(ErrNoMRCsFound).Msg("when handling MRC event, found no MRCs in OSM control plane namespace")
+		return ErrNoMRCsFound
+	}
+
+	if len(mrcList) > 2 {
+		log.Error().Err(ErrNumMRCExceedsMaxSupported).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrNumMRCExceedsMaxSupported)).
```

Comment:
Given this log line (and just as a general question), is there any reason this function shouldn't just query the cluster (or the informer cache) and return the MRCs found in the cluster and a (nillable) error?

Reply:
during the mrc_reconciler event that could be the case but during the validation phase this function needs to replace the current intent with the desired intent before running the validation.  Separating the query and validation logic lets us re-use the validation logic.

Reply:
Got it; makes sense!